ci-operator/step-registry/aro-hcp/provision/azure-login/aro-hcp-provision-azure-login-commands.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+# Create an ARO HCP Cluster + Node pool using bicep.
+set -o errexit
+set -o nounset
+set -o pipefail
+
+CREDENTIALS_DIR="/var/run/${TENANT_ID}"
+if [[ ! -d "${CREDENTIALS_DIR}" ]]; then
+  echo "Credentials directory '${CREDENTIALS_DIR}' does not exist." >&2
+  exit 1
+fi
+
+
+CLIENT_ID=$(cat "${CREDENTIALS_DIR}/client-id")
+CLIENT_SECRET=$(cat "${CREDENTIALS_DIR}/client-secret")
+TENANT_ID=$(cat "${CREDENTIALS_DIR}/tenant")
+
+cat > "${SHARED_DIR}/az-login.sh" <<EOF
+#!/bin/bash
+set -euo pipefail
+az login --service-principal -u '${CLIENT_ID}' -p '${CLIENT_SECRET}' --tenant '${TENANT_ID}' --output none
```

Comment:
This seems slightly overkill. I think we can handle copy-pasting a login step (2 commands) in a couple places. Dynamically generating shell scripts to run them later is going to really bite you when it breaks. Don't bother.